src/main/java/org/testng/Assert.java

```diff
@@ -169,6 +169,271 @@ private static void assertArrayEquals(Object actual, Object expected, String mes
+   * an AssertionError is thrown.
+   *
+   * @param actual   the actual value
+   * @param expected the expected value
+   */
+  static public void assertEquals(float[] actual, float[] expected) {
+    assertEquals(actual, expected, "");
+  }
+
+  /**
+   * Asserts that two arrays contain the same elements in the same order. If they do not,
+   * an AssertionError, with the given message, is thrown.
+   *
+   * @param actual   the actual value
+   * @param expected the expected value
+   * @param message  the assertion error message
+   */
+  static public void assertEquals(float[] actual, float[] expected, String message) {
+    if (checkRefEqualityAndLength(actual, expected, message)) return;
+
+    for (int i = 0; i < expected.length; i++) {
+      if (expected[i] != actual[i]) {
+        failOnArrayMismatch(actual, expected, i, message);
+      }
+    }
+  }
+
+  /**
+   * Asserts that two arrays contain the same elements in the same order. If they do not,
+   * an AssertionError is thrown.
+   *
+   * @param actual   the actual value
+   * @param expected the expected value
+   */
+  static public void assertEquals(double[] actual, double[] expected) {
+    assertEquals(actual, expected, "");
+  }
+
+  /**
+   * Asserts that two arrays contain the same elements in the same order. If they do not,
+   * an AssertionError, with the given message, is thrown.
+   *
+   * @param actual   the actual value
+   * @param expected the expected value
+   * @param message  the assertion error message
+   */
+  static public void assertEquals(double[] actual, double[] expected, String message) {
+    if (checkRefEqualityAndLength(actual, expected, message)) return;
+
+    for (int i = 0; i < expected.length; i++) {
+      if (expected[i] != actual[i]) {
+        failOnArrayMismatch(actual, expected, i, message);
+      }
+    }
+  }
+
+  /**
+   * Asserts that two arrays contain the same elements in the same order. If they do not,
+   * an AssertionError is thrown.
+   *
+   * @param actual   the actual value
+   * @param expected the expected value
+   */
+  static public void assertEquals(long[] actual, long[] expected) {
+    assertEquals(actual, expected, "");
+  }
+
+  /**
+   * Asserts that two arrays contain the same elements in the same order. If they do not,
+   * an AssertionError, with the given message, is thrown.
+   *
+   * @param actual   the actual value
+   * @param expected the expected value
+   * @param message  the assertion error message
+   */
+  static public void assertEquals(long[] actual, long[] expected, String message) {
+    if (checkRefEqualityAndLength(actual, expected, message)) return;
+
+    for (int i = 0; i < expected.length; i++) {
+      if (expected[i] != actual[i]) {
+        failOnArrayMismatch(actual, expected, i, message);
+      }
+    }
+  }
+
+  static private void failOnArrayMismatch(Object actual, Object expected, int idx, String message) {
```

Comment:
Nice idea. We indeed doesn't have to use reflection for reporting error. I think we doesn't have to do "double templating" if we just convert argument to strings in place. Something like this:

```java
private static final String TEMPLATE = "Arrays differ firstly at element [%d]; Expected value is<%s>, Actual value was <%s>. ";

  static public void assertEquals(long[] actual, long[] expected, String message) {
    if (checkRefEqualityAndLength(actual, expected, message)) return;

    for (int i = 0; i < expected.length; i++) {
      if (expected[i] != actual[i]) {
        fail(String.format(TEMPLATE, i, Long.toString(expected[i]), Long.toString(actual[i])));
      }
    }
  }
```

What do you think?